Added possibility to switch off reusing the Node

statut opened the pull request:
Our project should run integration tests several times so that we need to reuse the ElasticSearch Node

alexcojocaru commented on src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java:
```diff
@@ -105,7 +110,7 @@ public void execute() throws MojoExecutionException
     protected ElasticsearchNode startNode(Settings settings) throws MojoExecutionException
     {
-        if (getNode() != null)
+        if (reuseElasticNode && getNode() != null)
```
Can you describe the use case you are trying to solve with this?

alexcojocaru replied:
My bad for not reading the description you added to the merge request; I understand now why you need this parameter.

If reuseElasticNode is false, then the node property is reset and you loose the reference to the previously started ES instance.

alexcojocaru commented on src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java:
```diff
@@ -54,6 +54,11 @@
     /**
+     * @parameter default-value=true
+     */
+    protected boolean reuseElasticNode = true;
```
Please add documentation and test(s).

statut commented:
I reverted my changes since there was no problem in reusing. Instead of this I added checking: getNode().isClose() before returning node in startNode() method and wrote test that reproduces bug: if we stopped the node and then calls startNode(), startNode() method returns the node from the context but that node is invalid.

alexcojocaru commented:
Looks good; thanks.

statut commented:
Could you, please, tell me, when these changes will be in the maven central? Thanks

alexcojocaru commented:
I am releasing a new version (2.3) of the plugin, it will be available in maven central in a couple of hours.

statut commented:
Бонус
Ют яЯ**":"'

сб, 19 нояб. 2016 г., 8:14 alexcojocaru <notifications@github.com>:

> I am releasing a new version (2.3) of the plugin, it will be available in
> maven central in a couple of hours.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/alexcojocaru/elasticsearch-maven-plugin/pull/21#issuecomment-261694377>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHm1zTrXRIog62wSNJdEaY02_rsZO9GAks5q_oW0gaJpZM4K1XMm>
> .
>

statut commented:
Thank you very much!
